### DISCUSSION TITLE
Resonance Structures not being fully computed

### DISCUSSION THREAD

**Discussion by gunabam** [rdkit/rdkit#4490] [Q&A] [answered]
Hi there, I am trying to calculate the potential resonance structures with rdkit. This is the code that I tried:
```
from rdkit import Chem
from rdkit.Chem.rdchem import ResonanceMolSupplier

mol = Chem.MolFromSmiles('[CH2+]C1=NC=CC=C1')
flags = Chem.KEKULE_ALL | Chem.ALLOW_INCOMPLETE_OCTETS | Chem.UNCONSTRAINED_CATIONS
suppl = ResonanceMolSupplier(mol, flags=flags)
for resMol in suppl:
    print(Chem.MolToSmiles(resMol))

```
And I get the following output:
```
[CH2+]c1ncccc1
[CH2+]c1ccccn1
C=c1cc[cH+]cn1
C=c1[cH+]cccn1
C=c1cccc[nH2+]1
```
However, when I try a different input such as `C=c1cccc[nH2+]1` (one of the potential resonance structures from the previous example), I am not getting the same structures. I get the following:
```
C=C1C=CC=C[NH2+]1
```
Is there something missing from my code? Thank you in advance for your help!

**Comment by ptosco** [comment, marked as answer]
@gunabam This is a bug and I have just filed an issue: #4491
Thanks for reporting - I'll look into fixing that when I find some time.
